### PR TITLE
Fix BoringSSL includes being not passed to dependent rules.

### PR DESCRIPTION
**Pull request by NikitaNikolaenko**

### CODE REVIEW

File: BUILD.boringssl.bazel [hunk 1]
```diff
@@ -0,0 +1,29 @@
+# Builds and outputs crypto and ssl static libs (.lib or .a).
+# Doesn't output headers of those rules!!!
+filegroup(
+    name = "boringssl_built_libs",
+    srcs = [
+        "@boringssl//:crypto",
+        "@boringssl//:ssl",
+    ],
+)
```
Comment (benh) [inline BUILD.boringssl.bazel]: Have you ever tried just putting `@boringssl//:crypto` and `@boringssl//:ssl` in `srcs` of `boringssl_lib` below? Does `filegroup` somehow cause it to do something different? If you've never tried it I'd love to know what is different.

File: BUILD.bazel [hunk 2]
```diff
@@ -115,9 +70,8 @@ cmake(
             "libcurl.lib",
         ],
     }),
-    visibility = ["//visibility:private"],
     deps = [
-        "//:boringssl_lib",
+        "@boringssl_with_bazel_rules_curl_build_file//:boringssl_lib",
```
Comment (benh) [inline BUILD.bazel:65]: What happened to the comment we had worked on together that explained why we were doing this whole dance? Let's please put it in!

**Review by benh — APPROVED** [review]
This looks good but it seems like we lost a very important comment! And now that you moved things into another file (which I like!) the comment at the top of `BUILD.bazel` should be updated.

File: BUILD.boringssl.bazel (hunk 1 above)
Comment (NikitaNikolaenko) [reply BUILD.boringssl.bazel]: It works!

File: BUILD.bazel (hunk 2 above)
Comment (NikitaNikolaenko) [reply BUILD.bazel:65]: Fixed

**Pull request merged**